base/compiler/ssair/passes.jl

```diff
@@ -723,53 +723,45 @@ function perform_lifting!(compact::IncrementalCompact,
 end
 
 function lift_svec_ref!(compact::IncrementalCompact, idx::Int, stmt::Expr)
-    if length(stmt.args) != 4
-        return
-    end
+    length(stmt.args) != 4 && return
 
     vec = stmt.args[3]
     val = stmt.args[4]
     valT = argextype(val, compact)
     (isa(valT, Const) && isa(valT.val, Int)) || return
     valI = valT.val::Int
-    (1 <= valI) || return
+    valI >= 1 || return
 
     if isa(vec, SimpleVector)
-        if valI <= length(val)
-            compact[idx] = vec[valI]
-        end
-        return
-    end
-
-    if isa(vec, SSAValue)
+        valI <= length(vec) || return
+        compact[idx] = QuoteNode(vec[valI])
```

Comment:
@vtjnash I don't quite understand why this needs `QuoteNode` (per https://github.com/JuliaLang/julia/pull/45062#discussion_r968884654). It's probably really simple, but could you explain?

Reply:
It may need a quote node since the right side is an arbitrary value and the left side expects an instruction

Reply:
usually we use `quoted` here rather than `QuoteNode` to avoid quoting things that don't need actual QuotNodes